docs/usage.md

```diff
@@ -402,6 +402,67 @@ Other options can be used to help *git-changelog* retrieving
 the latest entry from your changelog: `--version-regex`
 and `--marker-line`.
 
+
+## Configuration files
+
+Project-wise, permanent configuration of *git-changelog* is possible.
+By default, *git-changelog* will search for the existence a suitable configuration
+in the ``pyproject.toml`` file or otherwise, the following configuration files 
+in this particular order:
+    * ``.git-changelog.toml``
+    * ``config/git-changelog.toml``
+    * ``.config/git-changelog.toml``
+    * ``~/.config/git-changelog.toml``
+
+The use of a configuration file can be disabled or overridden with the ``--config-file``
+option.
+To disable the configuration file, pass ``no``, ``None``, ``false``, or ``0``:
+
+```bash
+git-changelog --config-file no
+```
+
+To override the configuration file, pass the path to the new file:
+
+```bash
+git-changelog --config-file $HOME/.custom-git-changelog-config
+```
+
+The configuration file must be written in TOML language, and may take values
+for most of the command line options:
+
+```toml
+bump-latest = false
+convention = 'basic'
+in-place = false
+marker-line = '<!-- insertion marker -->'
+output = 'output.log'
+parse-refs = false
+parse-trailers = false
+repository = '.'
+sections = ''
```

Comment:
Empty string and `None` are equivalent. I can remove the empty string in the example, I don't think it is very relevant.

Reply:
Yeah I'd prefer to remove support for the empty string or empty array. They're either an error on the user-side (it makes no sense to hide all sections), or it's confusing (empty = unset?).

Reply:
Okay, it seems toml does not accept null/nil. Basically, the setting should be absent. I think all settings should have some `None` mapping, so at least `'none'` would be nice to support here.

WDYT?

Reply:
Yeah I saw that too. Well, I think it's better to unset options rather than special-casing `'none'`. We can revisit later if needed. Lets keep things simple for now.

Reply:
I've removed the possibility of `'none'` here. This will need a revisit if, at some point, a hierarchy of configs is to be implemented (so as to be able to revert to default if a lower-priority config set it to something else).